pyfar/dsp/dsp.py

```diff
@@ -227,6 +227,169 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+    the indexes of the samples given in `length` denote the first sample of
+    the windowing which is non-zero and the first being one. For the right
+    side of a symmetric window and for ``shape='right'``, the samples denote
+    the last sample being one and the last being non-zero.
+
+    This function calls `~scipy.signal.windows.get_window` to create the
+    window.
+    Window types:
+    - ``scipy.signal.windows.boxcar``
+    - ``scipy.signal.windows.triang``
+    - ``scipy.signal.windows.blackman``
+    - ``scipy.signal.windows.hamming``
+    - ``scipy.signal.windows.hann``
+    - ``scipy.signal.windows.bartlett``
+    - ``scipy.signal.windows.flattop``
+    - ``scipy.signal.windows.parzen``
+    - ``scipy.signal.windows.bohman``
+    - ``scipy.signal.windows.blackmanharris``
+    - ``scipy.signal.windows.nuttall``
+    - ``scipy.signal.windows.barthann``
+    - ``scipy.signal.windows.kaiser`` (needs beta)
+    - ``scipy.signal.windows.gaussian`` (needs standard deviation)
+    - ``scipy.signal.windows.general_gaussian`` (needs power, width)
+    - ``scipy.signal.windows.dpss`` (needs normalized half-bandwidth)
+    - ``scipy.signal.windows.chebwin`` (needs attenuation)
+    - ``scipy.signal.windows.exponential`` (needs center, decay scale)
+    - ``scipy.signal.windows.tukey`` (needs taper fraction)
+    - ``scipy.signal.windows.taylor`` (needs number of constant sidelobes,
+      sidelobe level)
+    If the window requires no parameters, then `window` can be a string.
+    If the window requires parameters, then `window` must be a tuple
+    with the first argument the string name of the window, and the next
+    arguments the needed parameters.
+    If `window` is a floating point number, it is interpreted as the beta
+    parameter of the `~scipy.signal.windows.kaiser` window.
+    """
+    # Check input
+    if not isinstance(signal, pyfar.Signal):
+        raise TypeError("The parameter signal has to be of type: Signal.")
+    if shape not in ('symmetric', 'left', 'right'):
+        raise ValueError(
+            "The parameter shape has to be 'symmetric', 'left' or 'right'.")
+    if not isinstance(truncate, bool):
+        raise TypeError("The parameter truncate has to be of type: bool.")
+    if not isinstance(length, (list, type(None))):
+        raise TypeError(
+            "The parameter length has to be of type list or None.")
+
+    # Check length
+    if length is not None:
+        if length != sorted(length):
+            raise ValueError("Values in length need to be in ascending order.")
+        # Convert to samples
+        if unit == 's':
+            length = [round(li*signal.sampling_rate) for li in length]
+        elif unit == 'ms':
+            length = [round(li*signal.sampling_rate/1e3) for li in length]
+        elif unit != 'samples':
+            raise ValueError(f"unit is {unit} but has to be"
+                             f" 'samples', 's' or 'ms'.")
```

Comment:
This would make it more flexible and we also do this in `pyfar.signals`:
```suggestion
        elif unit == 'samples':
            length = [int(li) for li in length]
        else:
            raise ValueError(f"unit is {unit} but has to be"
                             f" 'samples', 's' or 'ms'.")
```

Reply:
thx